src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java

```diff
@@ -93,10 +120,117 @@ private String createDescription() {
+    }
+
+    private PublicationInstance extractPublicationInstance(DataciteResponse dataciteResponse) throws
+            InvalidPageTypeException {
+        if (JOURNAL_CONTENT.equals(getPublicationType(dataciteResponse))) {
+            DataciteContainer container = dataciteResponse.getContainer();
+            String issue = Optional.ofNullable(container.getIssue()).orElse(null);
+            String volume = Optional.ofNullable(container.getVolume()).orElse(null);
+            return new JournalArticle.Builder()
+                    .withArticleNumber(null)
+                    .withIssue(issue)
+                    .withPages(extractPages(container))
+                    .withVolume(volume)
+                    .withPeerReviewed(true)
+                    .build();
+        }
+        return null;
+    }
+
+    private Range extractPages(DataciteContainer container) {
+        return new Range.Builder()
+                .withBegin(container.getFirstPage())
+                .withEnd(container.getLastPage())
+                .build();
+    }
+
+    private PublicationContext extractPublicationContext(DataciteResponse dataciteResponse) throws
+            InvalidIssnException {
+        PublicationType type = getPublicationType(dataciteResponse);
+        if (nonNull(type) && type.equals(JOURNAL_CONTENT)) {
+
+            return new Journal.Builder()
+                    .withPrintIssn(extractPrintIssn(dataciteResponse))
+                    .withTitle(dataciteResponse.getContainer().getTitle())
+                    .withOnlineIssn(extractOnlineIssn(dataciteResponse))
+                    .withPeerReviewed(true)
+                    .withOpenAccess(extractOpenAccess(dataciteResponse))
+                    .withLevel(Level.NO_LEVEL)
+                    .build();
+        }
         return null;
     }
 
+    private String extractOnlineIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isOnlineIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+
+    }
+
+    private boolean isOnlineIssn(DataciteRelatedIdentifier identifier) {
+        return DataciteRelatedIdentifierType.getByCode(identifier.getRelatedIdentifierType())
+                .equals(DataciteRelatedIdentifierType.EISSN);
+    }
+
+    private String extractPrintIssn(DataciteResponse dataciteResponse) {
+        DataciteRelatedIdentifier result = extractPartOfRelations(dataciteResponse)
+                .stream()
+                .filter(this::isPrintIssn)
+                .collect(SingletonCollector.collectOrElse(null));
+        return nonNull(result) ? IssnCleaner.clean(result.getRelatedIdentifier()) : null;
+    }
+
+    private List<DataciteRelatedIdentifier> extractPartOfRelations(DataciteResponse dataciteResponse) {
```

Comment:
maybe extract**Is**PartOfRelations. This is maybe stupid but the first time I looked at the method, I was expecting a method that extracts a part (not the whole) of some relation set